packages/ui-kit/src/helpers/graphql/stories/MetricInput.stories.tsx

```diff
@@ -0,0 +1,45 @@
+import type { Meta, StoryObj } from '@storybook/react'
+import type { MetricInput } from '../index'
+
+export const MetricInputComponent: React.FC<MetricInput> = () => null
+
+const meta = {
```

Comment:
In GraphQL, we use `@oneOf` for this type. In TypeScript, it would be best defined as

```ts
type MetricInput
  = { average: AverageMetricInput }
  | { count: CountMetricInput }
  | { countDistinct: CountDistinctMetricInput }
  | { custom: CustomMetricInput }
  | { id: string }
  | { name: string }
  // …
```

Should we use this type instead? Is it ergonomic? Does Storybook support it?

Reply:
@oneOf generates the type like 
```
type MetricInput
  = { average: AverageMetricInput,  name: never... }
  ...
```
That is the problem. We need to find a better solution for that.